admin/partials/admin-monk-post-meta-box-field-render.php

```diff
@@ -80,13 +78,43 @@
 						</option>
 				<?php endif; endforeach; ?>
 			</select>
-			<button class="monk-submit-translation button"><?php _e( 'Ok', 'monk' ); ?></button>
-			<a class="monk-cancel-submit-translation hide-if-no-js button-cancel"><?php _e( 'Cancel', 'monk' ); ?></a>
+			<button class="monk-submit-translation button"><?php esc_html_e( 'Ok', 'monk' ); ?></button>
+			<a class="monk-cancel-submit-translation hide-if-no-js button-cancel"><?php esc_html_e( 'Cancel', 'monk' ); ?></a>
 		<?php endif; ?>
 	</div>
 	<ul class="monk-translated-to">
 		<li>
 			<?php echo esc_html( $monk_languages[$post_default_language]['name'] ); ?>
+
+			<!-- 
+				Gives the option to alter the current post language
+				When the user select this feature,
+				the option containig translations is updated
+			-->
+			<a class="edit-post-status hide-if-no-js">
+				<span aria-hidden="true" class="monk-change-language">
+					<?php esc_html_e( 'Change', 'monk' ); ?>
+				</span>
+				<span class="screen-reader-text"><?php esc_html_e( 'Change current language', 'monk' ); ?></span>
+			</a>
+			<div class="monk-change-current-language">
+				<?php if ( $translation_counter !== count( $active_languages ) ) : ?>
+					<select name="monk_new_language">
```

Comment:
Remove `monk_post_language` hidden field.
Use this select field to maintain/change the post language value using  `[name="monk_post_language"]`.